es-space-mgmt/main/default/lwc/reservationDetailForm/__tests__/reservationDetailForm.test.js

```diff
@@ -0,0 +1,244 @@
+            is: ReservationDetailForm
+        });
+        const DATE_RANGES = [
+            { label: '1 Day', value: '1' },
+            { label: '7 Days', value: '7' },
+            { label: '30 Days', value: '30' },
+            { label: '60 Days', value: '60' },
+            { label: 'More than 60 Days', value: '61' }
+        ];
+        element.state = STATE;
+        element.city = CITY;
+        element.people = 50;
+        element.startdate = START_DATE;
+        element.totaldays = 15;
+        document.body.appendChild(element);
+        // Emit mock response from @wire
+        getMarketsByStateAdapter.emit(mockMarketsByStateRecords);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            const lighntingLayoutEl = element.shadowRoot.querySelector(
+                'lightning-layout'
+            );
+            const lighntingComboBoxEl = element.shadowRoot.querySelector(
+                'lightning-combobox'
+            );
+            const lighntingSliderEl = element.shadowRoot.querySelector(
+                'lightning-slider'
+            );
+            const lighntingInputEl = element.shadowRoot.querySelector(
+                'lightning-input'
+            );
+            const lighntingRadioGroupEl = element.shadowRoot.querySelector(
+                'lightning-radio-group'
+            );
+            expect(lighntingLayoutEl).not.toBeNull();
+            expect(lighntingComboBoxEl).not.toBeNull();
+            expect(lighntingComboBoxEl.options).toStrictEqual(CITY_OPTIONS);
+            expect(lighntingComboBoxEl.placeholder).toBe('Choose a market');
+            expect(lighntingSliderEl).not.toBeNull();
+            expect(lighntingInputEl).not.toBeNull();
+            expect(lighntingRadioGroupEl).not.toBeNull();
+            expect(lighntingRadioGroupEl.options).toMatchObject(DATE_RANGES);
+        });
+    });
+
+    it('getMarketsByState @wire return no data', () => {
+        const element = createElement('c-reservation-detail-form', {
+            is: ReservationDetailForm
+        });
+        element.state = STATE;
+        element.city = CITY;
+        element.people = 50;
+        element.startdate = START_DATE;
+        element.totaldays = 15;
+        document.body.appendChild(element);
+        // Emit mock response with 0 rows
+        getMarketsByStateAdapter.emit(mockMarketsByStateNoRecords);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            const lighntingLayoutEl = element.shadowRoot.querySelector(
```

Comment:
Most of these element tests don't belong in here, you tested them already earlier. Only thing would be the `options` check on the combobox.

Reply:
fixed